src/transformers/pipelines/text_classification.py

```diff
@@ -152,10 +152,16 @@ def __call__(self, *args, **kwargs):
 
             If `top_k` is used, one such dictionary is returned per label.
         """
-        result = super().__call__(*args, **kwargs)
+        if isinstance(inputs, str) or isinstance(inputs, dict):
+            inputs = (inputs,)
+        else:
+            inputs = tuple(
+                [inputs],
+            )  # noqa: C409
```

Comment:
If `inputs` is a list, isn't this equivalent to just `(inputs,)` as well?

Reply:
So the issue here is that Python will convert the inner list into a tuple, meaning that `[1, 2, 3]` as `tuple([1, 2, 3],)` is not `([1, 2, 3],)` but `(1, 2, 3)` which is wrong, and in this case just the first arg from `inputs` (the former `*args`) is used, while the rest are discarded and printed out as "Ignored args: (2, 3)"

Reply:
Maybe there's a better workaround formatting-wise to do that in a cleaner way? But I was unsure so I went with the safest approach to keep the current implementation using `*args` as is, with no breaking changes at all.

Reply:
Hm - are you sure? This is what I got:
```python
>>> inputs = [1, 2, 3]
>>> tuple([inputs],)
([1, 2, 3],)
>>> (inputs,)
([1, 2, 3],)
```

It looks like both approaches have the same output!

Reply:
Ah, I see - they differ if `inputs` is a nested list.

Reply:
True, I think that the `tuple([...],)` was what was confusing me, sorry for the misunderstanding, and thanks for the heads up!